From c38927aaad7a0b9be8353458ea2d06c8d8bb3e1f Mon Sep 17 00:00:00 2001
From: Richard Zhao <b20223@freescale.com>
Date: Tue, 17 Feb 2009 19:32:06 +0800
Subject: [PATCH] ENGR00107806-1 sgtl5000: use func to control spk amp shutdown

Becuase spk amp shutdown pin can be connected to PM gpo or cpu GPIO, I use a
function amp_enable in arch/arm/mach-xxx/mxxxx-3stack.c to controll shutdown.

Signed-off-by: Richard Zhao <b20223@freescale.com>
---
 arch/arm/plat-mxc/include/mach/mxc.h |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/arch/arm/plat-mxc/include/mach/mxc.h b/arch/arm/plat-mxc/include/mach/mxc.h
index fb5c4f0..89ddee9 100644
--- a/arch/arm/plat-mxc/include/mach/mxc.h
+++ b/arch/arm/plat-mxc/include/mach/mxc.h
@@ -302,11 +302,16 @@ struct mxc_sgtl5000_platform_data {
 
 	char *vddio_reg;
 	char *vdda_reg;
-	char *amp_gpo;
+	char *vddd_reg;
 	int vddio;	/* voltage of VDDIO (uv) */
 	int vdda;	/* voltage of vdda (uv) */
 	int vddd;	/* voldtage of vddd (uv), zero if not connected */
 	int sysclk;
+
+	int (*init)(void); /* board specific init */
+	int (*amp_enable)(int enable);
+	int (*finit)(void); /* board specific finit */
+	void *priv; /* used by board specific functions */
 };
 
 extern void mxc_wd_reset(void);
-- 
1.5.4.4

